pandas/tests/reductions/test_reductions.py

```diff
@@ -979,10 +980,10 @@ def test_any_all_object_dtype_missing(self, data, bool_agg_func):
             ([True, pd.NA, False], [[True, False], [True, False]]),
         ],
     )
-    def test_any_all_boolean_kleene_logic(
-        self, bool_agg_func, skipna, data, expected_data
+    def test_any_all_nullable_kleene_logic(
+        self, bool_agg_func, skipna, data, dtype, expected_data
     ):
-        ser = Series(data, dtype="boolean")
+        ser = Series(data, dtype="boolean").astype(dtype)
```

Comment:
why is this astyped and not the dtype passed to the Series constructor?

Reply:
Have changed the parameterization to avoid needing to do this 

The reason was that something like `Series([True, pd.NA, True], dtype="Int64")` raises. Though using just pure booleans doesn't raise, eg
`Series([True, True, True], dtype="Int64")` is fine. Does this inconsistency make sense, or should I open an issue?

Reply:
> Does this inconsistency make sense, or should I open an issue?

`pd.Series([True, np.nan, True], dtype="Int64")` works fine, but `pd.Series([True, None, True], dtype="Int64")` also raises `TypeError: object cannot be converted to an IntegerDtype`

There does seam to appear to be inconsistency here.